fvm/crypto/crypto_test.go

```diff
@@ -414,13 +421,387 @@ func TestVerifySignatureFromTransaction(t *testing.T) {
+						return fingerprint.Fingerprint(struct {
+							Payload           interface{}
+							PayloadSignatures interface{}
+						}{
+							tb.payloadCanonicalForm(),
+							[]interface{}{
+								// Expected canonical form of payload signature
+								struct {
+									SignerIndex uint
+									KeyID       uint
+									Signature   []byte
+								}{
+									SignerIndex: uint(tb.PayloadSignatures[0].SignerIndex),
+									KeyID:       uint(tb.PayloadSignatures[0].KeyIndex),
+									Signature:   tb.PayloadSignatures[0].Signature,
+								},
+							},
+						})
+					},
+					require: func(t *testing.T, sigOk bool, err error) {
+						require.NoError(t, err)
+						require.True(t, sigOk)
+					},
+				}, {
+					payloadInfo: []byte{0x0},
+					expectedEnvelopeSignatureCanonicalForm: func(tb transactionBodyScafold) []byte {
+						return fingerprint.Fingerprint(struct {
+							Payload           interface{}
+							PayloadSignatures interface{}
+						}{
+							tb.payloadCanonicalForm(),
+							[]interface{}{
+								// NON-expected canonical form of payload signature
+								struct {
+									SignerIndex uint
+									KeyID       uint
+									Signature   []byte
+									Info        []byte
+								}{
+									SignerIndex: uint(tb.PayloadSignatures[0].SignerIndex),
+									KeyID:       uint(tb.PayloadSignatures[0].KeyIndex),
+									Signature:   tb.PayloadSignatures[0].Signature,
+									Info:        []byte{0x0},
+								},
+							},
+						})
+					},
+					require: func(t *testing.T, sigOk bool, err error) {
+						require.NoError(t, err)
+						require.False(t, sigOk)
+					},
+				}, {
+					payloadInfo: []byte{0x0},
+					expectedEnvelopeSignatureCanonicalForm: func(tb transactionBodyScafold) []byte {
+						return fingerprint.Fingerprint(struct {
+							Payload           interface{}
+							PayloadSignatures interface{}
+						}{
+							tb.payloadCanonicalForm(),
+							[]interface{}{
+								// Expected canonical form of payload signature
+								struct {
+									SignerIndex uint
+									KeyID       uint
+									Signature   []byte
+								}{
+									SignerIndex: uint(tb.PayloadSignatures[0].SignerIndex),
+									KeyID:       uint(tb.PayloadSignatures[0].KeyIndex),
+									Signature:   tb.PayloadSignatures[0].Signature,
+								},
+							},
+						})
+					},
+					require: func(t *testing.T, sigOk bool, err error) {
+						require.NoError(t, err)
+						require.True(t, sigOk)
+					},
+				},
+			}
+
+			for _, c := range cases {
+
+				t.Run(fmt.Sprintf("auth scheme (payloadInfo): %v", c.payloadInfo), func(t *testing.T) {
+					seed := make([]byte, seedLength)
+					_, err := rand.Read(seed)
+					require.NoError(t, err)
+					sk, err := onflowCrypto.GeneratePrivateKey(s, seed)
+					require.NoError(t, err)
+
+					hasher, err := crypto.NewPrefixedHashing(h, flow.TransactionTagString)
+					require.NoError(t, err)
+
```

Comment:
If I get the test logic correctly, I think `transactionMessage` should include `c. payloadInfo`. I think we need to apply the tested value to the transaction signatures and then get the message:
```
transactionBody.PayloadSignatures[0].Info = c.payloadInfo
transactionBody.EnvelopeSignatures[0].Info = c.payloadInfo
transactionMessage := transactionBody.EnvelopeMessage()
```

Reply:
I believe this is resolved in my latest changes, please reopen if not